gradio/blocks.py

```diff
@@ -310,8 +312,91 @@ def iterate_over_children(children_list):
                     event_method = getattr(original_mapping[target], trigger)
                     event_method(fn=fn, **dependency)
 
+            # Allows some use of Interface-specific methods with loaded Spaces
+            blocks.predict = [fns[0]]
+            dependency = blocks.dependencies[0]
+            blocks.input_components = [blocks.blocks[i] for i in dependency["inputs"]]
+            blocks.output_components = [blocks.blocks[o] for o in dependency["outputs"]]
+
+        blocks.api_mode = True
         return blocks
 
+    def __call__(self, *params, fn_index=0):
+        """
+        Allows Blocks objects to be called as functions
+        Parameters:
+        *params: the parameters to pass to the function
+        fn_index: the index of the function to call (defaults to 0, which for Interfaces, is the default prediction function)
+        """
+        dependency = self.dependencies[fn_index]
+        block_fn = self.fns[fn_index]
+
+        if self.api_mode:
+            serialized_params = []
+            for i, input_id in enumerate(dependency["inputs"]):
+                block = self.blocks[input_id]
+                if getattr(block, "stateful", False):
+                    raise ValueError(
+                        "Cannot call Blocks object as a function if any of"
+                        " the inputs are stateful."
+                    )
+                else:
+                    serialized_input = block.serialize(params[i], True)
+                    serialized_params.append(serialized_input)
+        else:
+            serialized_params = params
+
+        processed_input = self.preprocess_data(fn_index, serialized_params, None)
+
+        if inspect.iscoroutinefunction(block_fn.fn):
+            raise ValueError(
```

Comment:
@abidlabs why not run async function in `asyncio.run()`, I think we should support async functions as well, or this feature is incomplete.

Reply:
The problem is that this approach wouldn't work inside of a jupyter notebook. You get this error: https://stackoverflow.com/questions/55409641/asyncio-run-cannot-be-called-from-a-running-event-loop

I wasn't sure how to fix it myself, so I left this case as unsupported for now

Reply:
hmm, I see, solved smt like [this](https://github.com/FarukOzderim/Memory-Vault/blob/master/src/__init__.py#L17-L41) in the past, it might help